src/pathwayvis/services/actions/actions.service.ts

```diff
@@ -43,7 +43,8 @@ export class ActionsService {
    * @return {types.Action} Action
    */
   public getAction(type: string): types.Action {
-    return _.first(actionsList.filter((action: types.Action) => action.type === type));
+    let actionListAux = actionsList.filter((action: types.Action) => action.type === type);
+    return ((actionListAux.length > 0) ? actionListAux[0] : []);
```

Comment:
What's the type of `actionListAux`? I think you should return `null` if the condition evaluates to false